daft/runners/pyrunner.py

```diff
@@ -111,12 +116,13 @@ def glob_paths_details(
                 0: partition,
             }
         )
-        return pset
+        return pset, fs
```

Comment:
Agreed that the caller should ideally be the one to ensure consistency here. I did it this way so that we only resolve the filesystem once while still ensuring that:
1. The filesystem is resolved in a Ray task when using the Ray runner, in order to deal with asymmetric configurations between the client and the cluster.
2. We don't have to launch a(nother) blocking Ray task for the Ray runner to resolve it, to keep Ray task overhead minimal.

I chose what I thought was the simplest option, which is to just tack it onto the first Ray task (globbing + file info scraping) that needs the filesystem. We should probably have a single Ray task for path globbing, file info gathering, and schema fetching, which is what I was originally going to refactor towards, but I didn't want to block the PR on that refactor of the `RunnerIO` API.

Reply:
Oh! I see... It does feel iffy that the Ray cluster would send its credentials back to the client though.

Isn't it sufficient for the caller (the driver) to provide either:

1. `fs=None` to all calls, i.e. it's the Runner's responsibility to create a default fs
2. `fs=user_provided_fs` to all calls, i.e. the Runner should use the injected filesystem

Reply:
>i.e. I think it's fine (and probably cleaner from a security perspective as well) to dynamically create the fs when None is provided, instead of passing credentials around in our system

I initially didn't go that route since that can lead to inconsistent credentials if the on-disk fsspec configuration is different between Ray nodes, e.g. if the user uploaded a config to the Ray head node but not the worker nodes. I suppose that we're not pinning the globbing + file info scraping task to the Ray head node yet, so this inconsistency can happen anyway.

As for the security point, that's certainly valid, although was never brought up as a concern when describing this pattern to Ray users, most likely because nearly all users were running their driver program on the Ray head node.

How about I revert it back to what you described, where each Ray task will automatically create a default fs if a user did not provide one, and we can look at this again if/when consolidating all of the pre-reading fs interactions into a single Ray task?

Reply:
@jaychia Reverted back to the old semantics of repeated filesystem discovery!